src/bcdata/BCData.F90

```diff
@@ -917,7 +917,7 @@ subroutine totalSubsonicInlet
               &some faces.")
 
           call returnFail("totalSubsonicInlet", errorMessage)
-          call mpi_barrier(ADflow_comm_world, ierr)
+         !  call mpi_barrier(ADflow_comm_world, ierr)
```

Comment:
I will delete this it is not needed anymore.  I can't see a reason we want a barrier instead of a fail flag here, because in the optimization loop on HPC, some procs will have the BC dataset and not others, so without the returnFail it will just hang on those procs when the BC normals are flipped.

Reply:
have you verified that this does work as intended now? Would the code get stuck at any other barriers until it reaches that mpi allreduce?

Reply:
I haven't seen any issues since making this change on my fork, but I can look for the barriers associated with the boundary conditions and see if any of them should be calling `returnFail` in the same fashion.